ui/src/components/VolumeMetricGraphCard.js

```diff
@@ -161,52 +161,59 @@ const MetricGraphCard = (props) => {
 
   // slot[0] => timestamp
   // slot[1] => value
-  const volumeUsageData = volumeUsedOperated?.map((slot) => {
+
+  const volumeUsageData = volumeUsageOperated?.map((slot) => {
+    return {
+      date: fromUnixTimestampToDate(slot[0]),
+      y: slot[1] === null ? null : Math.round(slot[1] * 100),
+    };
+  });
+
+  const volumeLatencyWriteData = volumeLatencyWriteOperated?.map((slot) => {
     return {
-      date: new Date(slot[0] * 1000), // convert from the RFC 3339 to time in JS
-      y:
-        slot[1] === null
-          ? null
-          : Math.round((slot[1] / volumeStorageCapacity) * 100),
+      date: fromUnixTimestampToDate(slot[0]),
+      write: slot[1] === null ? null : Math.round(toMicro(slot[1])),
```

Comment:
I fact, I think we can ask Prometheus to do the conversion instead of calling `toMicro`.

For instance, using this query:

```javascript
  const volumeLatencyWriteQuery = `sum(irate(node_disk_write_time_seconds_total[5m]) / irate(node_disk_writes_completed_total[5m])) by (instance, device) * 1000000`;
```

To be tested, but I think it works (and if so, same logic can be used for `toMega`).
IMHO, we should let Prometheus do as much work as possible.

Reply:
> we should let Prometheus do as much work as possible

Should we? Why add load on a critical server-side component when a user's computer is able to do the computation without too much effort? What about the change of scale from the UI (not in scope for now, but wouldn't be absurd)?

Reply:
Because it's Prometheus purpose to deal with those data efficiently (it's a database), so I expect it to be more optimized for this kind of task than some JS code in the browser.
I don't think the multiplication is an heavy operation for Prometheus (otherwise I would be really worried).

If we don't want to load Prometheus, than we should simply fetch raw data and do everything (sampling, aggregation, …) with JS code, no?

> What about the change of scale from the UI (not in scope for now, but wouldn't be absurd)?

The data on the chart are already refreshed regularly, so we would simply update the query we use on the fly.
This won't require extra requests.

Reply:
Thanks for the explanation, it makes sense indeed. More than a database (some of them are not built with the same purpose in mind), Prometheus is optimized for distributed queries with advanced computations such as the ones you mentioned, so let's use it :+1:!